standard-controller/src/main/java/io/enmasse/controller/standard/AddressProbeClient.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.controller.standard;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.ConcurrentHashSet;
+import io.vertx.proton.ProtonClient;
+import io.vertx.proton.ProtonClientOptions;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonReceiver;
+import io.vertx.proton.ProtonSender;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.messaging.Accepted;
+import org.apache.qpid.proton.amqp.messaging.AmqpValue;
+import org.apache.qpid.proton.message.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class AddressProbeClient implements AutoCloseable {
+    private static final Logger log = LoggerFactory.getLogger(AddressProbeClient.class);
+
+    private final Vertx vertx;
+    private final String containerId;
+
+    private Context context;
+    private ProtonConnection connection;
+    private final List<ProtonSender> senders = new ArrayList<>();
+    private final List<ProtonReceiver> receivers = new ArrayList<>();
+
+    public AddressProbeClient(Vertx vertx, String containerId) {
+        this.vertx = vertx;
+        this.containerId = containerId;
+    }
+
+    public void connect(String host, int port, ProtonClientOptions clientOptions, CompletableFuture<Void> promise) {
+        if (connection != null) {
+            log.debug("Already connected");
+            promise.complete(null);
+            return;
+        }
+
+        ProtonClient client = ProtonClient.create(vertx);
+        log.debug("Connecting to {}:{}", host, port);
+        client.connect(clientOptions, host, port, result -> {
```

Comment:
Perhaps add a connection properties to help identity the work of the heath check in the router log.

Reply:
I believe the containerId is set to `standard-controller-healthcheck`. Is this sufficient, or should it be set in properties as well?

Reply:
Container id is sufficient.